src/main/java/hudson/maven/AbstractMavenProcessFactory.java

```diff
@@ -234,8 +238,87 @@ private Object writeReplace() {
         public String call() throws IOException {
             return System.getProperty("file.encoding");
         }
-    }    
+    }
+
+    private static final class LinkedChannelCleaner extends Channel.Listener {
+        final WeakReference<LinkedChannel> weakChannel;
+
+        private LinkedChannelCleaner(LinkedChannel channel) {
+            this.weakChannel = new WeakReference<LinkedChannel>(channel);
+        }
+
+        public void onClosed(Channel upstreamChannel, IOException cause) {
+            LinkedChannel channel = weakChannel.get();
+            if (channel != null) {
+                channel.terminateNow(cause);
+            }
+        }
+    }
+
+    private static class LinkedChannel extends Channel {
+        private static final Logger LOGGER = Logger.getLogger(LinkedChannel.class.getName());
+
+        protected final LinkedChannelCleaner cleanerListener;
+
+        protected final Channel upstreamChannel;
+
+        public LinkedChannel(VirtualChannel upstreamChannel, String name, ExecutorService exec, InputStream is, OutputStream os, OutputStream header) throws IOException {
+            super(name, exec, is, os, header);
+
+            if(upstreamChannel instanceof Channel) {
+                this.cleanerListener = new LinkedChannelCleaner(this);
+                this.upstreamChannel = (Channel) upstreamChannel;
+                this.upstreamChannel.addListener(cleanerListener);
+            } else {
+                this.cleanerListener = null;
+                this.upstreamChannel = null;
+            }
+        }
+
+        public void terminateNow(IOException e) {
+            super.terminate(e);
+        }
+        
+        /**
+         * copy/paste of Channels.forProcess
```

Comment:
`hudson.remoting` package is in a signed jar file, so you will be SOoL if you try that